syncstorage/storage/sql/__init__.py

```diff
@@ -764,12 +766,13 @@ def _purge_expired_bsos(self, grace_period=0, max_per_loop=1000):
         }
 
     def _purge_expired_batches(self, grace_period=0, max_per_loop=1000):
+        self._maybe_optimize_table_before_purge("OPTIMIZE_BATCHES_TABLE")
         return self._purge_items_loop("batch_uploads", "PURGE_BATCHES", {
             "lifetime": BATCH_LIFETIME,
             "grace": grace_period,
             "maxitems": max_per_loop,
         })
-        self._maybe_optimize_table("OPTIMIZE_BATCHES_TABLE")
+        self._maybe_optimize_table_after_purge("OPTIMIZE_BATCHES_TABLE")
```

Comment:
Unreachable code after `return`? (I think I missed this when you originally added `_maybe_optimize_table`, apologies!)

Reply:
Nice catch!  I'm astonished this didn't fail linting, digging in now to see if there's a problem with the linter...

Reply:
Looks like flake8 doesn't catch this as an error, but pylint does.  There's a separate project going on to migrate some of these repos over to use pylint, so I'll just fix the error here and leave the broader linting question alone for now...